Feature/exp64f

sk1er52 opened the pull request:
--gtest_filter=*exp*:*Exp*:*EXP*:

dkurt committed:
Cross compile

Build with Syntacore Clang

Run on self-hosted

Disable OpenCL

summary

Run baseline

filter from PR body

dkurt commented:
Failed tests:
```
[  FAILED  ] Core_Pow.accuracy
[  FAILED  ] Core_Pow.special
[  FAILED  ] Core_Exp/ElemWiseTest.accuracy/0, where GetParam() = 16-byte object <C0-4B DD-DA 2A-00 00-00 10-4C DD-DA 2A-00 00-00>
```

shyama7004 committed:
fix meanStdDev overflow for large images

asmorkalov committed:
Do not rely on cv namespace in HAL.

asmorkalov committed:
Merge pull request #26878 from asmorkalov:as/respect_namespace_hal

Do not rely on cv namespace in HAL

Malacath-92 committed:
Merge pull request #26879 from Malacath-92:4.x

Add missing include in gislandmodel.hpp #26879

Add `<exception>`, `<string>`, and `<cstddef>` includes to `gislandmodel.hpp` which are required due to the usage of `std::exception_ptr`, `std::string`, and `size_t` in this header.

Notably one of those causes a build error on recent versions of Xcode: #26780 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [N/A] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [N/A] The feature is well documented and sample code can be built with the project CMake

dkurt commented:
```
[ RUN      ] Core_Pow.accuracy
/home/runner/work/opencv/opencv/opencv/modules/ts/src/ts.cpp:612: Failure
Failed

	failure reason: Bad accuracy
	test case #5
	seed: 00000000000c5a61
-----------------------------------
	LOG:
output: Too big difference (=3.29233e-12 > 2.52385e-13) at (0, 1)
input array 0 type=64fC4, size=(148, 335)
ref output array 0 type=64fC4, size=(148, 335)
test_case_idx = 5

-----------------------------------
	CONSOLE: .
-----------------------------------

[  FAILED  ] Core_Pow.accuracy (47 ms)
```
```
[ RUN      ] Core_Exp/ElemWiseTest.accuracy/0, where GetParam() = 16-byte object <C0-4B 3E-E6 2A-00 00-00 10-4C 3E-E6 2A-00 00-00>
/home/runner/work/opencv/opencv/opencv/modules/core/test/test_arithm.cpp:1577: Failure
too big relative difference (3.3428088463414267e-12 > 9.9999999999999998e-13) between 64fC3 2-dim (1474 x 2) 'dst0' and 'dst' at [1, 0].
- dst0:
4.268084052887259e-08, 2.638258089929395e-10;
*4.213665845841915e+23*, 7.45185338876216e-24;
1.636533272696507e-08, 3.703296477991213e+29.
- dst:
4.26808405285973e-08, 2.638258089929394e-10;
*4.21366584582783e+23*, 7.451853388762197e-24;
1.636533272696506e-08, 3.703296477991205e+29.


src[0] ~ 64fC3 2-dim (1474 x 2)
testCase #0

[  FAILED  ] Core_Exp/ElemWiseTest.accuracy/0, where GetParam() = 16-byte object <C0-4B 3E-E6 2A-00 00-00 10-4C 3E-E6 2A-00 00-00> (2 ms)
```

dkurt commented on modules/core/perf/perf_math.cpp:
```diff
@@ -36,6 +38,40 @@ PERF_TEST_P(VectorLength, phase64f, testing::Values(128, 1000, 128*1024, 512*102
+
+    declare.in(src).out(dst);
+
+    RNG rng(getTickCount());
```
use `cv::randu`

dkurt commented on modules/core/perf/perf_math.cpp:
```diff
@@ -36,6 +38,40 @@ PERF_TEST_P(VectorLength, phase64f, testing::Values(128, 1000, 128*1024, 512*102
     SANITY_CHECK(angle, 5e-5);
 }
 
+PERF_TEST_P(VectorLength, exp64f_performance, testing::Values(128, 1000, 128*1024, 512*1024, 1024*1024))
```
Please try `Size_MatType`

dkurt commented on 3rdparty/hal_rvv/hal_rvv_1p0/exp64f.hpp:
```diff
@@ -0,0 +1,72 @@
+    if (n <= 0) return 0;
+    
+    const double LN2 = 0.6931471805599453;
+    const double invLN2 = 1.4426950408889634;
```
make static

dkurt commented on 3rdparty/hal_rvv/hal_rvv_1p0/exp64f.hpp:
```diff
@@ -0,0 +1,72 @@
+inline int exp64f(const double* src, double* dst, int n) {
+    if (n <= 0) return 0;
+    
+    const double LN2 = 0.6931471805599453;
```
Use `CV_LOG2`

dkurt replied:
@sk1er52 accuracy now better. Now one one test failed: 
```
[ RUN      ] Core_Pow.accuracy
/home/runner/work/opencv/opencv/opencv/modules/ts/src/ts.cpp:612: Failure
Failed

	failure reason: Bad accuracy
	test case #5
	seed: 00000000000c5a61
-----------------------------------
	LOG:
output: Too big difference (=2.89632e-13 > 2.52385e-13) at (0, 445)
input array 0 type=64fC4, size=(148, 335)
ref output array 0 type=64fC4, size=(148, 335)
test_case_idx = 5

-----------------------------------
	CONSOLE: .
-----------------------------------

[  FAILED  ] Core_Pow.accuracy (39 ms)
```
Let's replace inverse LN2 to more precise version too

sk1er52 replied:
@dkurt one test failed again, more precise version didn't help, I will try a new version with 1 more term in the expansion of the Taylor series, maybe then there will be enough accuracy

sk1er52 replied:
@dkurt 
now all the tests have passed, but there was no acceleration in the benchmark at the institute with the size of the m1 register, with m8 about 7.5-8.0, it seems like such data should be
![image](https://github.com/user-attachments/assets/49469056-6016-48e6-85a5-f68a8abd143a)

Kumataro committed:
Merge pull request #26859 from Kumataro:fix26858

imgcodecs:gif: support IMREAD_UNCHANGED and IMREAD_GRAYSCALE #26859

Close #26858 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

amane-ame committed:
Merge pull request #26804 from amane-ame:norm_hal_rvv

Add RISC-V HAL implementation for cv::norm and cv::normalize #26804

This patch implements `cv::norm` with norm types `NORM_INF/NORM_L1/NORM_L2/NORM_L2SQR` and `Mat::convertTo` function in RVV_HAL using native intrinsic, optimizing the performance for `cv::norm(src)`, `cv::norm(src1, src2)`, and `cv::normalize(src)` with data types `8UC1/8UC4/32FC1`.

`cv::normalize` also calls `minMaxIdx`, #26789 implements RVV_HAL for this.

Tested on MUSE-PI for both gcc 14.2 and clang 20.0.

```
$ opencv_test_core --gtest_filter="*Norm*"
$ opencv_perf_core --gtest_filter="*norm*" --perf_min_samples=300 --perf_force_samples=300
```

The head of the perf table is shown below since the table is too long.

View the full perf table here: [hal_rvv_norm.pdf](https://github.com/user-attachments/files/18468255/hal_rvv_norm.pdf)

<img width="1304" alt="Untitled" src="https://github.com/user-attachments/assets/3550b671-6d96-4db3-8b5b-d4cb241da650" />

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

shyama7004 committed:
replace null literals with nullptr; optimize setidentity with std::fill for cv_64fc1

dkurt commented on modules/core/perf/perf_math.cpp:
```diff
@@ -36,6 +38,68 @@ PERF_TEST_P(VectorLength, phase64f, testing::Values(128, 1000, 128*1024, 512*102
+
+    TEST_CYCLE_N(200)
+    {
+        cv::cv_hal_rvv::exp64f(src.data(), dst.data(), static_cast<int>(length));//hal::exp64f(src.data(), dst.data(), static_cast<int>(length));//cv_hal_rvv::exp64f(src.data(), dst.data(), static_cast<int>(length));
```
`cv::exp`

dkurt replied:
https://docs.opencv.org/5.x/d2/de8/group__core__array.html#ga3e10108e2162c338f1b848af619f39e5

dkurt commented on 3rdparty/hal_rvv/hal_rvv_1p0/exp64f.hpp:
```diff
@@ -0,0 +1,64 @@
+        vfloat64m8_t term = frac_part;
+        vfloat64m8_t res = __riscv_vfadd_vf_f64m8(frac_part, 1.0, vl);
+
+        const long double factor_term [10] {0.166666666666666666666666666666, 0.041666666666666666666666666666, 0.008333333333333333333333333333, 0.001388888888888888888888888888,
```
1) 1.0 / N
2) make static

dkurt commented on modules/core/perf/opencl/perf_arithm.cpp:
```diff
@@ -1255,6 +1255,7 @@ OCL_PERF_TEST_P(ReduceAccFixture, Reduce,
     SANITY_CHECK_NOTHING();
 }
 
+
```
let's revert

dkurt commented on modules/core/perf/perf_math.cpp:
```diff
@@ -36,6 +38,65 @@ PERF_TEST_P(VectorLength, phase64f, testing::Values(128, 1000, 128*1024, 512*102
+    // }
+
+    SANITY_CHECK(dst, 5e-10);
+}
```
let's delete this test

dkurt commented on modules/core/src/mathfuncs_core.simd.hpp:
```diff
@@ -3,6 +3,7 @@
 // of this distribution and at http://opencv.org/license.html.
 
 #include "mathfuncs.hpp"
+#include <iostream>
```
revert

dkurt commented on 3rdparty/hal_rvv/hal_rvv.hpp:
```diff
@@ -24,6 +24,7 @@
 #include "hal_rvv_1p0/mean.hpp" // core
 #include "hal_rvv_1p0/minmax.hpp" // core
 #include "hal_rvv_1p0/atan.hpp" // core
+#include "hal_rvv_1p0/exp64f.hpp"
```
please rename to `exp.hpp`

dkurt commented on 3rdparty/hal_rvv/hal_rvv_1p0/exp64f.hpp:
rename to `exp.hpp`

asmorkalov committed:
Merge pull request #26867 from shyama7004:fix-meanStdDev

fix meanStdDev overflow for large images

dkurt commented on 3rdparty/hal_rvv/hal_rvv_1p0/exp.hpp:
```diff
@@ -0,0 +1,64 @@
+#define OPENCV_HAL_RVV_EXP64F_HPP_INCLUDED
+
+#include <riscv_vector.h>
+#include <iostream>
```
not needed include

dkurt commented on modules/core/perf/perf_arithm.cpp:
```diff
@@ -798,4 +798,27 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/ , PatchNaNsFixture,
 
+//////////////EXP64F////////////
+typedef Size_MatType Exp64f;
+PERF_TEST_P(Exp64f, exp64f_performance, 
```
Let's name like this:
```cpp
typedef Size_MatType ExpFixture;
PERF_TEST_P(ExpFixture, Exp
```

dkurt commented on modules/core/perf/perf_arithm.cpp:
```diff
@@ -798,4 +798,27 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/ , PatchNaNsFixture,
+        cv::exp(src, dst);
+
+    SANITY_CHECK_NOTHING();
+    }
```
`SANITY_CHECK_NOTHING` should be outside of the test loop:
```cpp
    TEST_CYCLE() cv::exp(src, dst);
    SANITY_CHECK_NOTHING();
```

dkurt commented on modules/core/perf/perf_math.cpp:
```diff
@@ -1,4 +1,5 @@
 #include "perf_precomp.hpp"
+#include "hal_rvv.hpp"
```
revert

dkurt commented on modules/core/perf/perf_math.cpp:
```diff
@@ -22,6 +22,7 @@ PERF_TEST_P(VectorLength, phase32f, testing::Values(128, 1000, 128*1024, 512*102
     SANITY_CHECK(angle, 5e-5);
 }
 
+
```
let's revert

dkurt commented on modules/core/perf/perf_arithm.cpp:
```diff
@@ -798,4 +798,27 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/ , PatchNaNsFixture,
+//////////////EXP64F////////////
+typedef Size_MatType ExpFixture;
+PERF_TEST_P(ExpFixture, Exp, 
+    testing::Combine(testing::Values(TYPICAL_MAT_SIZES), testing::Values(CV_64F)))
```
Please add `CV_32F` too for the future improvements

dkurt commented on 3rdparty/hal_rvv/hal_rvv.hpp:
```diff
@@ -24,6 +24,7 @@
 #include "hal_rvv_1p0/mean.hpp" // core
 #include "hal_rvv_1p0/minmax.hpp" // core
 #include "hal_rvv_1p0/atan.hpp" // core
+#include "hal_rvv_1p0/exp.hpp"
```
```
#include "hal_rvv_1p0/exp.hpp"  // core
```

sk1er52 committed:
ver 1.0

sk1er52 committed:
fix perf test

asmorkalov committed:
Merge pull request #26882 from shyama7004:nullptr-setidentity

replace null literals with nullptr and optimize setidentity with std::fill

shyama7004 committed:
fix wrong python type hints for imread

RoshniUG committed:
Merge pull request #26662 from RoshniUG:4.x

Update window_cocoa.mm #26662

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake
- [x ] Added reference to the original bug report (#26661).
- [x]Updated the code as per the reviewer's suggestion to use a ternary operator.
- [x] Verified that the feature is properly documented and can be built with CMake.

Kumataro committed:
Merge pull request #26889 from Kumataro:fix26877

doc: update supporting imgcodec format settings #26889

Close https://github.com/opencv/opencv/issues/26877

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

asmorkalov committed:
Merge pull request #26890 from shyama7004:type-hint

fix wrong python type hints for imread

MaximSmolskiy committed:
Refactor test for FileStorage Base64

Kumataro committed:
imgcodecs: tiff: refactor reading scanlines test

asmorkalov committed:
Merge pull request #26894 from Kumataro:fix26893

imgcodecs: tiff: refactor Imgcodecs_Tiff_decode_Huge test

asmorkalov committed:
Merge pull request #26891 from MaximSmolskiy:refactor-test-for-filestorage-base64

Refactor test for FileStorage Base64

asmorkalov committed:
Use HAL for cv::mean function too.

shyama7004 committed:
fix hal_replacement typos

mshabunin committed:
RISC-V: error message in the toolchain file when compiler is not found

asmorkalov committed:
Merge pull request #26897 from shyama7004:typos-fix

fix hal_replacement typos

Kababey committed:
Update optical_flow.cpp

ref: push_back is changed to emplace_back in order to avoid unnecessary conversions [Scalar(r, g, b))] .

asmorkalov committed:
Move OpenVX implementation for FAST to HAL.

lve-gh committed:
Merge pull request #26884 from lve-gh:split8u_rvv_hal

[HAL] split8u RVV 1.0 #26884

### Pull Request Readiness Checklist
* Banana Pi BF3 (SpacemiT K1)
* Compiler: Syntacore Clang 18.1.4 (build 2024.12)
```
Geometric mean (ms)

                  Name of Test                   baseline  hal      hal
                                                    ui               vs
                                                                  baseline 
                                                                     ui
                                                                 (x-factor)
split::Size_Depth_Channels::(127x61, 8UC1, 2)     0.012   0.004     3.12   
split::Size_Depth_Channels::(127x61, 8UC1, 3)     0.019   0.006     2.91   
split::Size_Depth_Channels::(127x61, 8UC1, 4)     0.028   0.011     2.64   
split::Size_Depth_Channels::(127x61, 8UC1, 5)     0.067   0.033     2.02   
split::Size_Depth_Channels::(127x61, 8UC1, 6)     0.084   0.040     2.11   
split::Size_Depth_Channels::(127x61, 8UC1, 7)     0.103   0.055     1.88   
split::Size_Depth_Channels::(127x61, 8UC1, 8)     0.113   0.032     3.50   
split::Size_Depth_Channels::(640x480, 8UC1, 2)    0.454   0.179     2.54   
split::Size_Depth_Channels::(640x480, 8UC1, 3)    0.677   0.298     2.27   
split::Size_Depth_Channels::(640x480, 8UC1, 4)    0.901   0.410     2.20   
split::Size_Depth_Channels::(640x480, 8UC1, 5)    3.781   3.010     1.26   
split::Size_Depth_Channels::(640x480, 8UC1, 6)    4.886   4.009     1.22   
split::Size_Depth_Channels::(640x480, 8UC1, 7)    5.777   4.770     1.21   
split::Size_Depth_Channels::(640x480, 8UC1, 8)    4.596   1.330     3.46   
split::Size_Depth_Channels::(1280x720, 8UC1, 2)   1.377   0.709     1.94   
split::Size_Depth_Channels::(1280x720, 8UC1, 3)   2.091   1.034     2.02   
split::Size_Depth_Channels::(1280x720, 8UC1, 4)   2.744   1.573     1.74   
split::Size_Depth_Channels::(1280x720, 8UC1, 5)   9.542   6.284     1.52   
split::Size_Depth_Channels::(1280x720, 8UC1, 6)   11.114  7.850     1.42   
split::Size_Depth_Channels::(1280x720, 8UC1, 7)   14.083  11.879    1.19   
split::Size_Depth_Channels::(1280x720, 8UC1, 8)   13.524  3.865     3.50   
split::Size_Depth_Channels::(1920x1080, 8UC1, 2)  3.108   1.395     2.23   
split::Size_Depth_Channels::(1920x1080, 8UC1, 3)  4.659   2.128     2.19   
split::Size_Depth_Channels::(1920x1080, 8UC1, 4)  6.127   2.818     2.17   
split::Size_Depth_Channels::(1920x1080, 8UC1, 5)  26.733  16.625    1.61   
split::Size_Depth_Channels::(1920x1080, 8UC1, 6)  31.242  22.414    1.39   
split::Size_Depth_Channels::(1920x1080, 8UC1, 7)  35.968  27.658    1.30   
split::Size_Depth_Channels::(1920x1080, 8UC1, 8)  29.997  8.655     3.47
```
See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

shyama7004 committed:
Fix _DEBUG/NDEBUG handling across modules (#26151)

asmorkalov committed:
Merge pull request #26908 from shyama7004:_DEBUG/NDEBUG

Fix _DEBUG/NDEBUG handling across modules

asmorkalov committed:
Move OpenVX integrations to imgproc to OpenVX HAL

Covered functions:
- medianBlur
- Sobel
- Canny
- pyrDown
- BoxFilter
- equalizeHist
- GaussianBlur
- remap
- threshold

asmorkalov committed:
Merge pull request #26895 from asmorkalov:as/mean_hal

Use HAL for cv::mean function too

asmorkalov committed:
Switch to static instance of FastCV on Linux.

mshabunin committed:
Merge pull request #26915 from mshabunin:fix-png-be

Resolves #26913
Related(?): #25715 #26832

asmorkalov committed:
Merge pull request #26917 from asmorkalov:as/static_fastcv

Switch to static instance of FastCV

ConnorBaker committed:
cmake/OpenCVDetectCUDAUtils.cmake: use IN_LIST to avoid regex matching valid capabilities

asmorkalov committed:
Merge pull request #26920 from ConnorBaker:fix/cmake-matches-uses-regex

cmake/OpenCVDetectCUDAUtils.cmake: use IN_LIST to avoid regex matching valid capabilities

asmorkalov committed:
Merge pull request #26903 from asmorkalov:as/openvx_hal

Migrate remaning OpenVX integrations to OpenVX HAL (core) #26903

Tested with OpenVX 1.2 & 1.3 sample implementation.

Steps to build and test:
```
git clone git@github.com:KhronosGroup/OpenVX-sample-impl.git
cd OpenVX-sample-impl
python3 Build.py --os=Linux --conf=Release
cd ..
mkdir build
cmake -DWITH_OPENVX=ON -DOPENVX_ROOT=/mnt/Projects/Projects/OpenVX-sample-impl/install/Linux/x64/Release/ ../opencv
make -j8
```

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

asmorkalov committed:
Merge pull request #26898 from mshabunin:fix-riscv-toolchain

RISC-V: error message in the toolchain file when compiler is not found

asmorkalov committed:
Merge pull request #26902 from Kababey:patch-3

Update optical_flow.cpp

asmorkalov committed:
Merge pull request #26875 from asmorkalov:as/in_memory_models

Added trackers factory with pre-loaded dnn models #26875

Replaces https://github.com/opencv/opencv/pull/26295

Allows to substitute custom models or initialize tracker from in-memory model.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

asmorkalov committed:
Merge pull request #26907 from asmorkalov:as/openvx_hal_features2d

Migrate remaning OpenVX integrations to OpenVX HAL (features2d)

shyama7004 committed:
Fix assertion in cv2.sampsonDistance

asmorkalov committed:
Merge pull request #26921 from shyama7004:sampsonDistance

Fix assertion in cv2.sampsonDistance

asmorkalov committed:
Squeeze several Windows pipelines into one with jobs.

shyama7004 committed:
fix minor issues in calib3d Docs

asmorkalov committed:
Merge pull request #26928 from shyama7004:doxFix

fix minor issues in calib3d Docs

asmorkalov committed:
Merge pull request #26911 from asmorkalov:as/openvx_hal_imgproc

Migrate remaning OpenVX integrations to OpenVX HAL (imgproc)

asmorkalov committed:
Merge pull request #26927 from asmorkalov:as/squeeze_windows

Squeeze several Windows pipelines into one with jobs

shyama7004 committed:
replace tostring() with tobytes

asmorkalov committed:
Removed Android test as it's broken for now.

asmorkalov committed:
Merge pull request #26933 from asmorkalov:as/drop_android_test

Removed Android test as it's broken for now

asmorkalov committed:
Merge pull request #26932 from shyama7004:deprecationFix

replace tostring() with tobytes()

kyler1cartesis committed:
Merge pull request #26887 from kyler1cartesis:4.x

invSqrt SIMD_SCALABLE implementation & HAL tests refactoring #26887

Enable CV_SIMD_SCALABLE for invSqrt.

* Banana Pi BF3 (SpacemiT K1) RISC-V
* Compiler: Syntacore Clang 18.1.4 (build 2024.12)

```
Geometric mean (ms)

                Name of Test                  baseline   simd      simd   
                                                       scalable  scalable
                                                                    vs
                                                                 baseline
                                                                (x-factor)
InvSqrtf::InvSqrtfFixture::(127x61, 32FC1)     0.163    0.051      3.23   
InvSqrtf::InvSqrtfFixture::(127x61, 64FC1)     0.241    0.103      2.35   
InvSqrtf::InvSqrtfFixture::(640x480, 32FC1)    6.460    1.893      3.41   
InvSqrtf::InvSqrtfFixture::(640x480, 64FC1)    9.687    3.999      2.42   
InvSqrtf::InvSqrtfFixture::(1280x720, 32FC1)   19.292   5.701      3.38   
InvSqrtf::InvSqrtfFixture::(1280x720, 64FC1)   29.452   11.963     2.46   
InvSqrtf::InvSqrtfFixture::(1920x1080, 32FC1)  43.326   12.805     3.38   
InvSqrtf::InvSqrtfFixture::(1920x1080, 64FC1)  65.566   26.881     2.44
```

vrabaud committed:
Bump openjp2 to v2.5.3

This should quiet some fuzzer bugs

asmorkalov committed:
Merge pull request #26942 from vrabaud:png_leak

Bump openjp2 to v2.5.3

dkurt committed:
Merge pull request #26923 from dkurt:merge_rvv_opt

Further optimization of cv::merge RVV HAL for 8U and 16S #26923

### Pull Request Readiness Checklist


* Banana Pi BF3 (SpacemiT K1) RISC-V
* Compiler: Syntacore Clang 18.1.4 (build 2024.12)

```
Geometric mean (ms)

                     Name of Test                       baseline   pr       pr
                                                         merge              vs    
                                                                         baseline
                                                                          merge
                                                                        (x-factor)
merge::Size_SrcDepth_DstChannels::(127x61, 8UC1, 2)      0.013   0.003     3.76   
merge::Size_SrcDepth_DstChannels::(127x61, 8UC1, 3)      0.020   0.006     3.46   
merge::Size_SrcDepth_DstChannels::(127x61, 8UC1, 4)      0.026   0.010     2.61   
merge::Size_SrcDepth_DstChannels::(127x61, 8UC1, 5)      0.043   0.028     1.56   
merge::Size_SrcDepth_DstChannels::(127x61, 8UC1, 6)      0.054   0.035     1.53   
merge::Size_SrcDepth_DstChannels::(127x61, 8UC1, 7)      0.065   0.050     1.30   
merge::Size_SrcDepth_DstChannels::(127x61, 8UC1, 8)      0.070   0.036     1.95   
merge::Size_SrcDepth_DstChannels::(127x61, 16SC1, 2)     0.015   0.008     1.82   
merge::Size_SrcDepth_DstChannels::(127x61, 16SC1, 3)     0.022   0.015     1.48   
merge::Size_SrcDepth_DstChannels::(127x61, 16SC1, 4)     0.029   0.018     1.63   
merge::Size_SrcDepth_DstChannels::(127x61, 16SC1, 5)     0.067   0.044     1.54   
merge::Size_SrcDepth_DstChannels::(127x61, 16SC1, 6)     0.088   0.056     1.58   
merge::Size_SrcDepth_DstChannels::(127x61, 16SC1, 7)     0.104   0.076     1.38   
merge::Size_SrcDepth_DstChannels::(127x61, 16SC1, 8)     0.116   0.065     1.79   
merge::Size_SrcDepth_DstChannels::(640x480, 8UC1, 2)     0.421   0.176     2.39   
merge::Size_SrcDepth_DstChannels::(640x480, 8UC1, 3)     0.792   0.284     2.79   
merge::Size_SrcDepth_DstChannels::(640x480, 8UC1, 4)     1.090   0.370     2.95   
merge::Size_SrcDepth_DstChannels::(640x480, 8UC1, 5)     1.835   1.399     1.31   
merge::Size_SrcDepth_DstChannels::(640x480, 8UC1, 6)     2.389   1.776     1.35   
merge::Size_SrcDepth_DstChannels::(640x480, 8UC1, 7)     3.000   2.471     1.21   
merge::Size_SrcDepth_DstChannels::(640x480, 8UC1, 8)     3.178   2.104     1.51   
merge::Size_SrcDepth_DstChannels::(640x480, 16SC1, 2)    0.490   0.377     1.30   
merge::Size_SrcDepth_DstChannels::(640x480, 16SC1, 3)    1.348   0.602     2.24   
merge::Size_SrcDepth_DstChannels::(640x480, 16SC1, 4)    1.827   0.813     2.25   
merge::Size_SrcDepth_DstChannels::(640x480, 16SC1, 5)    3.283   2.692     1.22   
merge::Size_SrcDepth_DstChannels::(640x480, 16SC1, 6)    4.922   3.334     1.48   
merge::Size_SrcDepth_DstChannels::(640x480, 16SC1, 7)    5.725   4.399     1.30   
merge::Size_SrcDepth_DstChannels::(640x480, 16SC1, 8)    6.278   4.748     1.32   
merge::Size_SrcDepth_DstChannels::(1280x720, 8UC1, 2)    1.267   0.603     2.10   
merge::Size_SrcDepth_DstChannels::(1280x720, 8UC1, 3)    2.394   0.934     2.56   
merge::Size_SrcDepth_DstChannels::(1280x720, 8UC1, 4)    3.236   1.434     2.26   
merge::Size_SrcDepth_DstChannels::(1280x720, 8UC1, 5)    5.398   4.345     1.24   
merge::Size_SrcDepth_DstChannels::(1280x720, 8UC1, 6)    7.127   5.459     1.31   
merge::Size_SrcDepth_DstChannels::(1280x720, 8UC1, 7)    8.590   7.298     1.18   
merge::Size_SrcDepth_DstChannels::(1280x720, 8UC1, 8)    9.360   6.152     1.52   
merge::Size_SrcDepth_DstChannels::(1280x720, 16SC1, 2)   1.482   1.242     1.19   
merge::Size_SrcDepth_DstChannels::(1280x720, 16SC1, 3)   4.008   1.817     2.21   
merge::Size_SrcDepth_DstChannels::(1280x720, 16SC1, 4)   6.079   2.468     2.46   
merge::Size_SrcDepth_DstChannels::(1280x720, 16SC1, 5)   11.300  8.644     1.31   
merge::Size_SrcDepth_DstChannels::(1280x720, 16SC1, 6)   15.125  12.126    1.25   
merge::Size_SrcDepth_DstChannels::(1280x720, 16SC1, 7)   17.555  14.804    1.19   
merge::Size_SrcDepth_DstChannels::(1280x720, 16SC1, 8)   18.890  14.163    1.33   
merge::Size_SrcDepth_DstChannels::(1920x1080, 8UC1, 2)   2.910   1.326     2.19   
merge::Size_SrcDepth_DstChannels::(1920x1080, 8UC1, 3)   5.351   1.997     2.68   
merge::Size_SrcDepth_DstChannels::(1920x1080, 8UC1, 4)   7.290   2.629     2.77   
merge::Size_SrcDepth_DstChannels::(1920x1080, 8UC1, 5)   12.426  9.611     1.29   
merge::Size_SrcDepth_DstChannels::(1920x1080, 8UC1, 6)   16.453  12.162    1.35   
merge::Size_SrcDepth_DstChannels::(1920x1080, 8UC1, 7)   19.420  16.190    1.20   
merge::Size_SrcDepth_DstChannels::(1920x1080, 8UC1, 8)   20.588  13.699    1.50   
merge::Size_SrcDepth_DstChannels::(1920x1080, 16SC1, 2)  3.400   2.640     1.29   
merge::Size_SrcDepth_DstChannels::(1920x1080, 16SC1, 3)  8.986   3.952     2.27   
merge::Size_SrcDepth_DstChannels::(1920x1080, 16SC1, 4)  11.972  5.273     2.27   
merge::Size_SrcDepth_DstChannels::(1920x1080, 16SC1, 5)  20.544  17.996    1.14   
merge::Size_SrcDepth_DstChannels::(1920x1080, 16SC1, 6)  28.677  22.086    1.30   
merge::Size_SrcDepth_DstChannels::(1920x1080, 16SC1, 7)  32.958  27.713    1.19   
merge::Size_SrcDepth_DstChannels::(1920x1080, 16SC1, 8)  36.499  27.439    1.33
```

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

dkurt committed:
SIMD_SCALABLE for sqrt

dkurt committed:
exp SIMD_SCALABLE

dkurt committed:
Keep only UI for faster integration
